### ISSUE TITLE
Code action to convert between struct types

### ISSUE THREAD

**Issue by oriongonza** [rust-lang/rust-analyzer#19666]
```rs
struct S {} // Convert into unit struct
struct S; // Convert into normal struct

struct S { field: String }; // Convert into tuple struct
struct S(String); // Convert into normal struct

struct S(String, u64); // Convert into normal struct
struct S { 
  field_1: String, // These can be renamed easily later on
  field_2: u64
}
```
I think that this would aid in refactoring in a big way, especially for tuple -> named structs.

It's a pain to fix them manually.

**Comment by roife** [comment]
We already have this assist. Note that you need to place the cursor on the name of the `struct` to activate it. It would be better to expand the trigger area to allow more people to discover this feature.

**Comment by Veykril** [comment]
I'd expect this assist to trigger  from the struct/visibility token up to the end of the name. The entire definition is way too annoying. Stuff that affects the entire definition of something imo should be triggerable only on the "signature" of the thing.

**Comment by ada4a** [comment]
> I'd expect this assist to trigger from the struct/visibility token up to the end of the name

AFAICT this was implemented in #19901, so this can be closed I think?
